kythe/java/com/google/devtools/kythe/platform/shared/filesystem/ByteBufferByteChannel.java

```diff
@@ -55,7 +60,11 @@ public int read(ByteBuffer dst) throws IOException {
       return -1;
     }
     wanted = (int) Math.min(possible, wanted);
-    dst.put(buffer, (int) position, wanted);
+    try {
+      dst.put(buffer.get(), (int) position, wanted);
+    } catch (InterruptedException | ExecutionException exc) {
```

Comment:
We could be "smarter" here and use get-with-timeout, and return 0 if it times out or interrupted. WDYT?

Reply:
I don't want to be in the choosing arbitrary timeouts business here.

Reply:
Ok, but at some level, all channels need to do that. In fact, you are choosing an arbitrary timeout -- +inf. But OK.

Reply:
It's buried pretty far in the stack, doesn't have easy access to command line flags, can be explicitly canceled via `close` and any non-infinite timeout we might choose is going to be too short for some case or indistinguishable from infinite anyway.